api/rebac/v1/check.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package api.rebac.v1;
+import "google/api/annotations.proto";
+option go_package = "ciam-rebac/api/rebac/v1;v1";
+option java_multiple_files = true;
+option java_package = "api.rebac.v1";
+
+service Check {
+	rpc CheckPermission (CheckPermissionRequest) returns (CheckPermissionResponse) {
+		option (google.api.http) = {
+			post: "/api/authz/v1/check"
+			body: "*"
+		};
+	};
+}
+
+message CheckPermissionRequest {
+	string subject = 1;
+	string subjectid = 2;
+	string operation = 3;
+	string resourcetype = 4;
+	string resourceid = 5;
+	bool auditlog = 6;
+}
+
+message CheckPermissionResponse {
+	bool result = 1;
```

Comment:
Let's use an enum value instead of a bool, even if there are just two values for now. But it lets us add more in the future (such as for support for ABAC where context may be missing in the call) while having a fallback "unknown" value. See: https://protobuf.dev/programming-guides/proto3/#enum

Something like

```
enum CheckResult {
  UNSPECIFIED = 0;
  ALLOWED = 1;
  FORBIDDEN = 2;
}
```

Values are just examples, there may be better ideas here.

Reply:
@akoserwal pointed out that this suggestion is somewhat conflicting with my comment above. So a better suggestion might be

```
enum Allowed {
  ALLOWED_UNSPECIFIED = 0;
  ALLOWED_TRUE = 1;
  ALLOWED_FALSE = 2;
  // e.g. future could have, ALLOWED_CONDITIONAL = 3;
```

Just an example. Open to other ideas!